src/psyclone/nemo.py

```diff
@@ -0,0 +1,822 @@
+        #                      Int_Literal_Constant('1', None)]), None)
+        limits_list = ctrl[0].items[1][1]
+        self._start = str(limits_list[0])
+        self._stop = str(limits_list[1])
+        if len(limits_list) == 3:
+            self._step = str(limits_list[2])
+        else:
+            # Default loop increment is 1
+            self._step = "1"
+
+        # Is this loop body a kernel?
+        if NemoKern.match(self._ast):
+            self.addchild(NemoKern(self._ast, parent=self))
+            return
+        # It's not - walk on down the AST...
+        self.process_nodes(self, self._ast.content, self._ast)
+
+    def __str__(self):
+        result = ("NemoLoop[" + self._loop_type + "]: " + self._variable_name +
+                  "=" + ",".join([self._start, self._stop, self._step]) + "\n")
+        for entity in self._children:
+            result += str(entity) + "\n"
+        result += "EndLoop"
+        return result
+
+    @property
+    def kernel(self):
+        ''' Returns the kernel object if one is associated with this loop,
+        None otherwise. '''
+        kernels = self.walk(self.children, NemoKern)
+        if kernels:
+            # TODO cope with case where loop contains >1 kernel (e.g.
+            # following loop fusion)
+            return kernels[0]
+        return None
+
+
+class NemoImplicitLoop(NemoLoop):
+    '''
+    Class representing an implicit loop in NEMO (i.e. using Fortran array
+    syntax).
+
+    :param ast: the part of the fparser2 AST representing the loop
+    :type ast: :py:class:`fparser.two.Fortran2003.Assignment_Stmt`
+    :param parent: the parent of this Loop in the PSyclone AST
+    '''
+    def __init__(self, ast, parent=None):
+        from fparser.common.readfortran import FortranStringReader
+        Loop.__init__(self, parent=parent,
+                      valid_loop_types=VALID_LOOP_TYPES+["unknown"])
+        # Keep a ptr to the corresponding node in the AST
+        self._ast = ast
+
+        # Get a reference to the name-space manager
+        name_space_manager = NameSpaceFactory().create()
+
+        # Find all uses of array syntax in the statement
+        subsections = walk_ast(ast.items, [Fortran2003.Section_Subscript_List])
+        # A Section_Subscript_List is a tuple with each item the
+        # array-index expressions for the corresponding dimension of the array.
+        for idx, item in enumerate(subsections[0].items):
+            if isinstance(item, Fortran2003.Subscript_Triplet):
+                # If an array index is a Subscript_Triplet then it is a range
+                # and thus we need an explicit loop for this dimension.
+                outermost_dim = idx
+
+        # TODO allow for implicit loops with specified bounds (e.g. 2:jpjm1)
+        self._start = 1
+        self._step = 1
+        if outermost_dim == 0:
+            # TODO ensure no name clash is possible with variables that
+            # already exist in the NEMO source.
```

Comment:
Good point. Please add to the list of issues.

Reply:
Done.